### PR TITLE
Allow configuration of chunksize parameter

### PR DESCRIPTION
**Pull request by stevvooe**
The code using values from the yaml package wasn't careful enought with the
possible incoming types. Turns out, it is just an int but we've made this
section somewhat bulletproof in case that package changes the behavior.

This code likely never worked. The configuration system should be decoupled
from the object instantiation.

Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

File: registry/storage/driver/s3/s3.go [hunk 1]
```diff
@@ -148,9 +149,23 @@ func FromParameters(parameters map[string]interface{}) (*Driver, error) {
...
+			if err != nil {
+				return nil, fmt.Errorf("chunksize parameter must be an integer, %v invalid", chunkSizeParam)
+			}
+			chunkSize = vv
+		case int64:
+			chunkSize = v
+		case int, uint, int32, uint32, uint64:
+			chunkSize = reflect.ValueOf(v).Convert(reflect.TypeOf(chunkSize)).Int()
+		default:
+			return nil, fmt.Errorf("invalid valud for chunksize: %#v", chunkSizeParam)
```
Comment (RichardScothern) [inline registry/storage/driver/s3/s3.go]: s/valud/value

**Comment by RichardScothern** [comment]
LGTM.

**Comment by dmcgowan** [comment]
LGTM

**Comment by dmp42** [comment]
LGTM

**Pull request merged**